src/main/java/io/debezium/connector/vitess/VitessMetadata.java

```diff
@@ -0,0 +1,201 @@
+                shardsToQuery = getVitessShardsFromTablets();
+            }
+            String randomShard = shardsToQuery.get(new Random().nextInt(shardsToQuery.size()));
+            LOGGER.info("Get tables from shard: {}", randomShard);
+            response = executeQuery(query, randomShard);
+        }
+        else {
+            query = String.format("SHOW TABLES FROM %s", config.getKeyspace());
+            response = executeQuery(query);
+        }
+        logResponse(response, query);
+        List<String> tables = getFlattenedRowsFromResponse(response);
+        LOGGER.info("All tables from keyspace {} are: {}", config.getKeyspace(), tables);
+        return tables;
+    }
+
+    private static void logResponse(Vtgate.ExecuteResponse response, String query) {
+        LOGGER.info("Got response: {} for query: {}", response, query);
+    }
+
+    private List<String> getVitessShards() {
+        String query = String.format("SHOW VITESS_SHARDS LIKE '%s/%%'", config.getKeyspace());
+        Vtgate.ExecuteResponse response = executeQuery(query);
+        logResponse(response, query);
+        List<String> rows = getFlattenedRowsFromResponse(response);
+        List<String> shards = rows.stream().map(fieldValue -> {
+            String[] parts = fieldValue.split("/");
+            assert parts != null && parts.length == 2 : String.format("Wrong field format: %s", fieldValue);
+            return parts[1];
+        }).collect(Collectors.toList());
+        return shards;
+    }
+
+    private List<String> getVitessShardsFromTablets() {
+        String query = "SHOW VITESS_TABLETS";
+        Vtgate.ExecuteResponse response = executeQuery(query);
+        // Do not log the response since there is no way to filter tablets: it includes all tablets of all shards of all keyspaces
+        List<List<String>> rowValues = getRowsFromResponse(response);
+        List<String> shards = VitessMetadata.getNonEmptyShards(rowValues, config.getKeyspace());
+        return shards;
+    }
+
+    private Vtgate.ExecuteResponse executeQuery(String query) {
+        return executeQuery(query, null);
+    }
+
+    @VisibleForTesting
+    protected Vtgate.ExecuteResponse executeQuery(String query, String shard) {
+        // Some tests need to be issue a shard-specific query, so make this visible
+        try (VitessReplicationConnection connection = new VitessReplicationConnection(config, null)) {
+            Vtgate.ExecuteResponse response;
+            if (shard != null) {
+                response = connection.execute(query, shard);
+            }
+            else {
+                response = connection.execute(query);
+            }
+            return response;
+        }
+        catch (Exception e) {
+            throw new RuntimeException(String.format("Unexpected error while running query: %s", query), e);
+        }
+    }
+
+    private static List<String> getFlattenedRowsFromResponse(Vtgate.ExecuteResponse response) {
```

Comment:
Is it possible to make the rest of the methods non-static too?

Reply:
It is possible, but I don't think we want to, since they don't rely on any instance variables. Static clearly conveys these methods do not rely on object state.